# SPDX-License-Identifier: GPL-2.0-or-later
# API 35 compatibility; does not change translator runtime code.
--- a/Android.bp
+++ b/Android.bp
@@ -1,3 +1,4 @@
+// Modified by Open Mobile Emulator for the API 35 build graph.
 // Copyright (C) 2023 The Android Open Source Project
 //
 // Licensed under the Apache License, Version 2.0 (the "License");
@@ -365,17 +366,15 @@
 }
 
 // This module is used as a workaround to define those deps of berberis_all_deps_defaults which are native_bridge only.
-ndk_translation_package {
+// OME API 35: use the native-bridge dependency names supported by A15.
+phony_rule {
     name: "berberis_all_deps_intermediate_package",
-    compile_multilib: "both",
-    native_bridge_supported: true,
-    native_bridge_deps: [
-        "berberis_hello_world",
-        "berberis_hello_world_static",
-        "berberis_ndk_program_tests",
-        "berberis_perf_tests_static",
-    ],
-    generate_build_files: false,
+    phony_deps: [
+        "berberis_hello_world.native_bridge",
+        "berberis_hello_world_static.native_bridge",
+        "berberis_ndk_program_tests.native_bridge",
+        "berberis_perf_tests_static.native_bridge",
+    ],
 }
 
 // region digitalis
@@ -467,53 +466,51 @@
 }
 
 // This module is used as a workaround to define those deps of berberis_all_deps_defaults which are native_bridge only.
-ndk_translation_package {
+// OME API 35: use the native-bridge dependency names supported by A15.
+phony_rule {
     name: "berberis_all_riscv64_to_x86_64_intermediate_package",
-    compile_multilib: "both",
-    native_bridge_supported: true,
-    native_bridge_deps: [
+    phony_deps: [
         // NATIVE_BRIDGE_PRODUCT_PACKAGES
-        "libnative_bridge_vdso",
-        "native_bridge_guest_app_process",
-        "native_bridge_guest_linker",
-        "libandroidicu",
-        "libcompiler_rt",
-        "libcrypto",
-        "libcutils",
-        "libdl",
-        "libdl_android",
-        "libicu",
-        "liblog",
-        "libm",
-        "libsqlite",
-        "libssl",
-        "libstdc++",
-        "libsync",
-        "libutils",
-        "libz",
+        "libnative_bridge_vdso.native_bridge",
+        "native_bridge_guest_app_process.native_bridge",
+        "native_bridge_guest_linker.native_bridge",
+        "libandroidicu.bootstrap.native_bridge",
+        "libcompiler_rt.native_bridge",
+        "libcrypto.native_bridge",
+        "libcutils.native_bridge",
+        "libdl.bootstrap.native_bridge",
+        "libdl_android.bootstrap.native_bridge",
+        "libicu.bootstrap.native_bridge",
+        "liblog.native_bridge",
+        "libm.bootstrap.native_bridge",
+        "libsqlite.native_bridge",
+        "libssl.native_bridge",
+        "libstdc++.native_bridge",
+        "libsync.native_bridge",
+        "libutils.native_bridge",
+        "libz.native_bridge",
         // NATIVE_BRIDGE_MODIFIED_GUEST_LIBS
-        "libnative_bridge_guest_libaaudio",
-        "libnative_bridge_guest_libamidi",
-        "libnative_bridge_guest_libandroid",
-        "libnative_bridge_guest_libandroid_runtime",
-        "libnative_bridge_guest_libbinder_ndk",
-        "//frameworks/libs/native_bridge_support/android_api/libc:libnative_bridge_guest_libc",
-        "libnative_bridge_guest_libcamera2ndk",
-        "libnative_bridge_guest_libEGL",
-        "libnative_bridge_guest_libGLESv1_CM",
-        "libnative_bridge_guest_libGLESv2",
-        "libnative_bridge_guest_libGLESv3",
-        "libnative_bridge_guest_libjnigraphics",
-        "libnative_bridge_guest_libmediandk",
-        "libnative_bridge_guest_libnativehelper",
-        "libnative_bridge_guest_libnativewindow",
-        "libnative_bridge_guest_libneuralnetworks",
-        "libnative_bridge_guest_libOpenMAXAL",
-        "libnative_bridge_guest_libOpenSLES",
-        "libnative_bridge_guest_libvulkan",
-        "libnative_bridge_guest_libwebviewchromium_plat_support",
-    ],
-    generate_build_files: false,
+        "libnative_bridge_guest_libaaudio.native_bridge",
+        "libnative_bridge_guest_libamidi.native_bridge",
+        "libnative_bridge_guest_libandroid.native_bridge",
+        "libnative_bridge_guest_libandroid_runtime.native_bridge",
+        "libnative_bridge_guest_libbinder_ndk.native_bridge",
+        "libnative_bridge_guest_libc.native_bridge",
+        "libnative_bridge_guest_libcamera2ndk.native_bridge",
+        "libnative_bridge_guest_libEGL.native_bridge",
+        "libnative_bridge_guest_libGLESv1_CM.native_bridge",
+        "libnative_bridge_guest_libGLESv2.native_bridge",
+        "libnative_bridge_guest_libGLESv3.native_bridge",
+        "libnative_bridge_guest_libjnigraphics.native_bridge",
+        "libnative_bridge_guest_libmediandk.native_bridge",
+        "libnative_bridge_guest_libnativehelper.native_bridge",
+        "libnative_bridge_guest_libnativewindow.native_bridge",
+        "libnative_bridge_guest_libneuralnetworks.native_bridge",
+        "libnative_bridge_guest_libOpenMAXAL.native_bridge",
+        "libnative_bridge_guest_libOpenSLES.native_bridge",
+        "libnative_bridge_guest_libvulkan.native_bridge",
+        "libnative_bridge_guest_libwebviewchromium_plat_support.native_bridge",
+    ],
 }
 
 // Note: Keep in sync with variables from `berberis_config.mk` and
